GOVERNANCE.md

```diff
@@ -263,3 +255,72 @@ approval.
 - Must be available to perform admin-related tasks (add a GitHub member, promote
   a GitHub user to "Owner", add someone to the GCP projects, etc.)
 - Must be responsible with the privileges granted to them.
+
+## Removing Contributors
+
+### Stepping Down
+
+Anyone who has reached any level within the cert-manager project can step down
+at any time.
+
+GitHub Members, Reviewers, Approvers, Maintainers and Admins should give notice
+of their intent to step down by raising an issue on the [community][] repo, so
+that their permissions can be revoked for security reasons.
+
+### Timing Out
+
+#### Maintainers
+
+A review of the [`MAINTAINERS.md`](./MAINTAINERS.md) file is performed every
+year by active current maintainers. During this review, the maintainers that
+have not been active in the last 18 months are asked whether they would like to
+become an emeritus maintainer and they are expected to respond within 30 days.
+
+If they do not respond, they will automatically be moved to emeritus status.
+
+#### Other Levels
+
+For security reasons, anyone at any level who isn't actively involved in the
+project for over a year is liable to be timed out upon review by an active
+maintainer.
+
+Anyone who is timed out can reach out to a maintainer to request to be
+reinstated to their previous level.
+
+There is no regularly scheduled check for any org level except for maintainer;
+timing out non-maintainers is ad-hoc.
+
+### Emeritus Status
+
+Anyone who has reached the "Maintainer" level will be added to the list of
+"emeriti" maintainers in `MAINTAINERS.md` upon stepping down or timing out.
+This is a marker of thanks for people who were involved in shaping the project.
+
+### Removal
+
+The cert-manager project abides by the code of conduct in [`CODE_OF_CONDUCT.md`](./CODE_OF_CONDUCT.md).
+
+Everyone interacting with the project must abide by this code of conduct, whether
+officially assigned one of the levels listed in this document or if simply
+interacting with the project (e.g. by joining a meeting or commenting on a
+GitHub issue). The same rules apply for members of the steering committee.
+
+If a committee with jurisdiction under the Code of Conduct recommends a person
+be removed from the project, then after the conclusion (if applicable) of any
+appeals, that person will be removed and will not be eligible to rejoin the
+project for at least one year from the date of their removal.
+
+#### Project Permissions and Removal
+
+If someone with an assigned level is undergoing an investigation by any
+committee with jurisdiction under the Code of Conduct, they should have all
+permissions removed temporarily while the investigation is underway.
```

Comment:
I understand the motivation/rationale for this, however, it may be contentious to take this action *immediately*...

I also wonder if this could be an issue wrt. privacy too - if a regular community member reports another regular community member, would we inform the accused party *immediately* that there is an investigation into them underway, or would we withhold this until an appropriate time/when required? With this policy in place, once a member with some form of privileges has a complaint filed against them, they'd be immediately aware due to permissions changes.

Reply:
I wonder if we should also explicitly list what is in-scope and out-of-scope for these permission changes too.. would we remove the member from GitHub orgs (to prevent them being able to edit/delete comments, manage labels on issues & PRs?).

I think the intent of these changes overall are fairly clear, but the devil is in the detail a bit, and it's particularly sensitive when it comes to code of conduct violations as these are usually, by definition, quite personal reports.

Reply:
I've tried to expand on this with more links to the CNCF documentation which I think states this well. Ultimately we'll always defer to the CNCF on this stuff.